agrona/src/main/java/org/agrona/concurrent/AtomicBuffer.java

```diff
@@ -16,16 +16,47 @@
 package org.agrona.concurrent;
 
 import org.agrona.MutableDirectBuffer;
+import org.agrona.SystemUtil;
+
+import static org.agrona.BitUtil.SIZE_OF_LONG;
 
 /**
  * Abstraction over a range of buffer types that allows type to be accessed with memory ordering semantics.
  */
 public interface AtomicBuffer extends MutableDirectBuffer
 {
+    /**
+     * Buffer alignment in bytes to ensure atomic word accesses.
+     */
+    int ALIGNMENT = SIZE_OF_LONG;
+
+    /**
+     * Name of the system property that specify if the alignment checks for atomic operations are strict. If the checks
+     * are strict then the {@link #verifyAlignment()} method will throw an exception if the underlying buffer is a
+     * {@code byte[]}.
+     * <p>
+     * The strictness of the check is platform-specific. For example on {@code x86} platform the check is disabled
+     * whereas on {@code aarch64} it is enabled.
+     */
+    String STRICT_ALIGNMENT_CHECKS_PROP_NAME = "agrona.strict.alignment.checks";
+
+    /**
+     * Should alignment checks for atomic operations be done or not. Controlled by the
+     * {@link #STRICT_ALIGNMENT_CHECKS_PROP_NAME} system property.
+     *
+     * @see AtomicBuffer#STRICT_ALIGNMENT_CHECKS_PROP_NAME
+     */
+    boolean STRICT_ALIGNMENT_CHECKS = "true".equals(
+        SystemUtil.getProperty(STRICT_ALIGNMENT_CHECKS_PROP_NAME, SystemUtil.isX86Arch() ? "false" : "true"));
+
     /**
      * Verify that the underlying buffer is correctly aligned to prevent word tearing and other ordering issues.
+     * <p>
```

Comment:
I think we need to go into a lot more detail here about the issues that can occur if accesses are unaligned, including the issues we have seen on Arm.  It should serve as a point of reference.  We should `@see` to the put/get methods and mention that indexes need to be aligned as well.

Reply:
Good point. I'll extend the JavaDoc here.

Reply:
Done.